experiments/integrated/fluxnet/data_tools.jl

```diff
@@ -54,81 +76,84 @@ function check_column(file_matrix::Matrix, column_name::String)
-    # Patch missing values with mean
-    if status == incomplete
-        num_missing = count(x -> x == -9999, col_dat)
-        percent_missing = 100.0 * num_missing / length(col_dat)
-        if percent_missing > 90.0
-            return DataColumn(col_dat, units, absent)
-        end
-        @info "Warning: Data for $column_name $(round(percent_missing,
-        sigdigits=3))% missing. Filling with mean value"
-        replace_missing_with_mean_by_value!(col_dat)
-        status = complete
-    end
-    return DataColumn(col_dat, units, status)
-end
+"""
+    filter_column(driver_data::Matrix, column_name::String, units::String)
+
+Checks the `driver_data` for the present of `column_name`; returns a 
+DataColumn struct with cleaned data and a complete status if present,
+returns an empty DataColumn with status absent if not.
 
-function VerifiedColumn(file_matrix::Matrix, column_name::String, units::String)
-    """Returns a DataColumn struct with units and status for a column read from 
-    the fluxnet data file. Checks to see if a quality control flag 
-    is available for the data column, and if so, replaces missing values with
-    the mean value per the QC flag."""
-    # Check that the data exists and read it in
-    col_dat, status = check_column(file_matrix, column_name)
+- Checks to see if a quality control flag 
+is available for the data column, and if so, replaces poor quality values with
+the mean value per the QC flag.
+ - Checks for missing data and replaces with the mean value
+- Returns an `absent` status if the column is not present.
+"""
+function filter_column(driver_data::Matrix, column_name::String, units::String)
+    # Check that the data exists and read it in if so
+    col_dat, status = column_status(driver_data, column_name)
+    # if it does not exist, exit
     if status == absent
         return DataColumn(reshape([], 0, 0), "", status)
-    end
-    # Check for a QC flag column
-    column_QC = column_name * "_QC"
-    QC_col, QCstatus = check_column(file_matrix, column_QC)
-    # If no QC flag, then fill and return the data column as is
-    if QCstatus != absent && minimum(QC_col) != 2
-        # Fill any missing data with the mean value per the QC flag
-        num_missing = count(x -> x > 1, QC_col)
-        if num_missing > 0
-            perc_missing = 100.0 * num_missing / length(col_dat)
-            @info "Warning: Data for $column_name $(
-            round(perc_missing, sigdigits=3))% missing. Filling with mean value"
-            replace_missing_with_mean!(col_dat, QC_col)
+    elseif status == complete # all data is present but some may be poor quality
+        # Check for a QC flag column and use it to clean data if possible
+        column_QC = column_name * "_QC"
+        QC_col, QCstatus = column_status(driver_data, column_QC)
+        # QC flag = [0,1] -> good data, [2,3] -> poorer data
+        num_poor = count(x -> x >= 2, QC_col)
+        if QCstatus != absent && num_poor > 0
```

Comment:
For some of the sites (including the Harvard site which is why the buildkite fails) there are QC columns showing that the entire column is poor quality. Then when we try to fill poor quality values there aren't any good indices to use. We just need a case to catch that case and mark the column as absent.

Reply:
If LW_IN is all poor quality and we mark as absent, how do we run the sim?

Reply:
Ah, before what I did was to mark the QC column as absent if it was all low quality instead of marking the data column as absent if the QC was all low quality to avoid this issue.

This just kind of ignores that all the data is low quality but I don't see too many options if the column is a required driver.